accounts/modelbackend.py

```diff
@@ -23,11 +23,32 @@
 
 
 class CustomModelBackend(ModelBackend):
-    def authenticate(self, username=None, password=None):
+
+    def authenticate(self, request, username=None, password=None, **kwargs):
         """ authenticate against case insensitive username """
-        try:
-            user = User.objects.get(username__iexact=username)
-            if user.check_password(password):
-                return user
-        except User.DoesNotExist:
-            return None
+
+        if '@' in username:
```

Comment:
The logic can be simplified and not have the same code twice, you're right. I did not do it because I tried to abstract the repeated part into a function but at the end it seemed to me the code was much less readable. However, if instead of abstracting into a function I just do one user lookup at the end I'll work the same. I committed that https://github.com/MTG/freesound/pull/863/commits/1fc30ee7084e04bd40650bc6b3a58b959139b7d0).

However, imagine the current situation: 

- User A has an account with username `user_a` and email `a@fs.org`
- User B has an account with username `a@fs.org` and email `whatever@fs.org`

If we check username first and only check email on the except case, User A won't be able to login using the email. Emails are valid usernames, but usernames are not valid emails if they don't have an @. The `if @ ...` clause is only to skip one db lookup when not using emails to log in.